src/lib.rs

```diff
@@ -1178,4 +1259,70 @@ mod tests {
         assert_eq!(hue, hue_after_mixing(Color::graytone(0.7)));
         assert_eq!(hue, hue_after_mixing(Color::white()));
     }
+
+    #[test]
+    fn color_scale_add_preserves_ordering() {
+        let mut color_scale = ColorScale::empty();
+
+        color_scale
+            .add_stop(Color::red(), Fraction::from(0.5))
+            .add_stop(Color::gray(), Fraction::from(0.0))
+            .add_stop(Color::blue(), Fraction::from(1.0));
+
+        assert_eq!(color_scale.color_stops.get(0).unwrap().color, Color::gray());
+        assert_eq!(color_scale.color_stops.get(1).unwrap().color, Color::red());
+        assert_eq!(color_scale.color_stops.get(2).unwrap().color, Color::blue());
+    }
+
+    #[test]
+    fn color_scale_empty_sample_none() {
+        let mix: Box<dyn Fn(&Color, &Color, Fraction) -> Color> =
+            Box::new(|c1: &Color, c2: &Color, f: Fraction| c1.mix::<Lab>(c2, f));
+
+        let color_scale = ColorScale::empty();
+
+        let color = color_scale.sample(Fraction::from(0.0), &mix);
```

Comment:
.. and it should still be `None` if one color is inserted, right? Would it make sense to test this as well?

Reply:
Yes, it should still be `None`. I'm adding a test too.